src/silo/query_engine/actions/tuple.cpp

```diff
@@ -363,10 +365,31 @@ std::vector<Tuple::ComparatorField> Tuple::getCompareFields(
 
 Tuple::Comparator Tuple::getComparator(
    const std::vector<silo::storage::ColumnMetadata>& columns_metadata,
-   const std::vector<OrderByField>& order_by_fields
+   const std::vector<OrderByField>& order_by_fields,
+   const std::optional<uint32_t>& randomize_seed
 ) {
    auto tuple_field_comparators =
       actions::Tuple::getCompareFields(columns_metadata, order_by_fields);
+   if (randomize_seed) {
+      const size_t seed = *randomize_seed;
+      return [tuple_field_comparators, seed](const Tuple& tuple1, const Tuple& tuple2) {
+         if (tuple1.compareLess(tuple2, tuple_field_comparators)) {
+            return true;
+         }
+         if (tuple2.compareLess(tuple1, tuple_field_comparators)) {
+            return false;
+         }
```

Comment:
To me, it's not quite clear yet how randomize is supposed to interact with "order by"? Do they overwrite each other? What happens if both are specified? Let's add a test that shows the behavior?

Reply:
Randomize will function only as a tiebreaker. Will add a unit test